node/lib/util/submodule_config_util.js

```diff
@@ -71,6 +71,34 @@ const CONFIG_FILE_NAME = "config";
 exports.modulesFileName = ".gitmodules";
 
 /**
+ * Return the relative path from the working directory of a submodule having
+ * the specified `subName` to its .git directory.
+ * @param {String} subName
+ */
+exports.computeRelativeGitDir = function (subName) {
```

Comment:
I'm a little concerned about this one.  We had discussed why git has configuration for submodules, and I think I have recalled why: because submodule paths are not unique across history.  So yesterday, my git-remote-hg might have pointed to github.com/felipec/git-remote-hg but today it points to github.com/mnauw/git-remote-hg.  How do we know which one was originally cloned into .git/modules?  That's what the config is for.

So I wonder if it would be better to check the config before manually generating this.